RELEASE_NOTES.rst

```diff
@@ -21,6 +21,60 @@
 
 .. towncrier release notes start
 
+Airflow 2.10.5 (2025-01-28)
+---------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- Ensure teardown tasks are executed when DAG run is set to failed
+
+  Previously when a DAG run was manually set to "failed" or to "success" state the terminal state was set to all tasks.
+  But this was a gap for cases when setup- and teardown tasks were defined: If teardown was used to clean-up infrastructure
+  or other resources, they were also skipped and thus resources could stay allocated.
+
+  As of now when setup tasks had been executed before and the DAG is manually set to "failed" or "success" then teardown
+  tasks are executed. Teardown tasks are skipped  if the setup was also skipped.
+
+  As a side effect this means if the DAG contains teardown tasks, then the manual marking of DAG as "failed" or "success"
+  will need to keep the DAG in running state to ensure that teardown tasks will be scheduled. They would not be scheduled
+  if the DAG is directly set to "failed" or "success". (#45530)
+
+
+Bug Fixes
+"""""""""
+
+- ``TriggerRule.ALWAYS`` cannot be utilized within a task-generated mapping, either in bare tasks (fixed in this PR) or mapped task groups (fixed in PR #44368). The issue with doing so, is that the task is immediately executed without waiting for the upstreams's mapping results, which certainly leads to failure of the task. This fix avoids it by raising an exception when it is detected during DAG parsing. (#44751)
+- Fix short circuit operator in mapped tasks. The operator did not work until now due to a bug in ``NotPreviouslySkippedDep``. Please note that at time of merging, this fix has been applied only for Airflow version > 2.10.4 and < 3, and should be ported to v3 after merging PR #44925. (#44912)
+- Fix pre-mature evaluation of tasks in mapped task group. The origins of the bug are in ``TriggerRuleDep``, when dealing with ``TriggerRule`` that is fastly triggered (i.e, ``ONE_FAILED``, ``ONE_SUCCESS`, or ``ONE_DONE``). Please note that at time of merging, this fix has been applied only for Airflow version > 2.10.4 and < 3, and should be ported to v3 after merging PR #40460. (#44937)
```

Comment:
Can you shorten the above changelogs or have them in significant changes if they are that important

Reply:
Sure, I'll do that